### PR TITLE
fix: links in markdown documentation

### PR DESCRIPTION
**Pull request by OnkarRuikar**
The PR fixes links in markdown docs.

### CODE REVIEW

File: RELEASE_NOTES.md [hunk 1]
```diff
@@ -6284,8 +6284,8 @@ This change was made in [#16415](https://github.com/mdn/browser-compat-data/pull
 
 ## Older Versions
 
-- [v4.x](./release_notes/v4.md)
```
Comment (queengooborg) [inline RELEASE_NOTES.md:6287]: Is there a reason for removing relative links?

Comment (teoli2003) [reply RELEASE_NOTES.md:6287]: We asked ourself the question last week: why `./release_notes/v4.md`  and not the shorter, relative `release_notes/v4.md` ? Is there a difference? There isn't on mdn/content but as these are files used by github, there may be.

Comment (teoli2003) [reply RELEASE_NOTES.md:6287]: But both cases are relative. An absolute link would start with `/` or `https:`.

Comment (queengooborg) [reply RELEASE_NOTES.md:6365]: Sorry, to clarify, I'm referring to _file_ relative (`./`, `../`), rather than root relative (`/`).

Comment (teoli2003) [reply RELEASE_NOTES.md:6365]: I don't understand:
- `[v4.x](./release_notes/v4.md)`
- `[v4.x](release_notes/v4.md)`

are the same, aren't they? Is there something I miss?

Comment (queengooborg) [reply RELEASE_NOTES.md:6365]: Not exactly.  A file that's on the project root is not a good example to demonstrate the issue, so let's break this down with a Markdown file that's located inside of a folder, say...`schemas/compat-data-schema.md`.

Suppose that inside of the compat data schema file, we include a link to the data guidelines, which are stored in `docs/data-guidelines` in the repo.  We add a link to the corresponding guideline, and we make the link relative to the project root -- so, `docs/data-guidelines/api.md#foo`.  On GitHub, this will correctly point to the `api.md` file inside of `docs/data-guidelines`.

Now, we go to view this file in a local Markdown viewer/editor (let's assume it's a WYSIWYG editor that doesn't have Git capabilities), and we attempt to click on the link and navigate to the file...and suddenly, we find that the link is broken.  This is because our local viewer/editor wouldn't know what to define as the root folder, so it wouldn't know what file to open.

Suppose that we used a file relative link instead -- so, `../docs/data-guidelines/api.md#foo`.  This will make both GitHub and our local viewer/editor happy, because it doesn't require knowledge of where the root folder is located.

_TLDR: On GitHub, both ways of linking to other files in Markdown work as intended.  This issue occurs when docs files are viewed and/or modified outside of GitHub._

Comment (OnkarRuikar) [reply RELEASE_NOTES.md:6365]: Are we changing all the links to relative (`./`)?

Comment (queengooborg) [reply RELEASE_NOTES.md:6365]: I personally think that would be better, yes.

Comment (teoli2003) [reply RELEASE_NOTES.md:6365]: I have no strong opinion here as long as it is consistent.

**Review by queengooborg — APPROVED** [review]
